corehq/apps/builds/models.py

```diff
@@ -219,6 +219,17 @@ def minor_release(self):
     def major_release(self):
         return self.version.split('.')[0]
 
+    def release_greater_than_or_equal_to(self, version):
+        if not self.version:
+            return False
+        from_major, from_minor, from_bugfix = map(int, version.split('.'))
+        major, minor, bugfix = map(int, self.version.split('.'))
+        if major != from_major:
+            return major > from_major
+        if minor != from_minor:
+            return minor > from_minor
+        return bugfix >= from_bugfix
```

Comment:
This comparison should probably use `distutils.version.LooseVersion` (look for existing usages - it's straightforward).  Then the code wouldn't need to assume all versions have two dots.  That should simplify the comparison.

Reply:
Your call, but that might even be small enough to move back in line

Reply:
thanks @esoergel

```
In [73]: LooseVersion('2.24') == LooseVersion('2.24.0')
Out[73]: False
```
That does not look right to me. Should not that be True?

Reply:
I think for that you'll need `StrictVersion`:
```
In [1]: from distutils.version import StrictVersion

In [2]: StrictVersion('2.4') == StrictVersion('2.4.0')
Out[2]: True
```

From the docs

> Version numbering for meticulous retentive and software idealists.
Implements the standard interface for version number classes as
described above.  A version number consists of two or three
dot-separated numeric components, with an optional "pre-release" tag
on the end.  The pre-release tag consists of the letter 'a' or 'b'
followed by a number.  If the numeric components of two version
numbers are equal, then one with a pre-release tag will always
be deemed earlier (lesser) than one without.

I do think our versioning schemes are probably consistent enough to use this.  Looks like HQ uses `LooseVersion` more than `StrictVersion` (36 vs 10 times), but strictness is generally a good idea when possible.

Reply:
thanks @esoergel